docs/notebooks/Calculating ground states on large scale systems with the quantum krylov subspace method.ipynb

```diff
@@ -0,0 +1,1037 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Calculating ground states on large scale systems: Quantum Krylov Subspaces"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1: Map problem to quantum native format\n",
+    "\n",
+    "Given a matrix H that we are interested in, subspace methods use a set of states as a basis for the construction of a smaller representation of H, which captures its properties of interest (e.g. lowest eigenvalue).\n",
+    "\n",
```

Comment:
The first sentence is a little awkward. I would suggest revising. I would also include at least one reference to a detailed explanation of the Krylov Subspace methods. One example is https://arxiv.org/pdf/2307.16279.pdf but any other would be good.

Reply:
Cool, I revised the first few sentences to introduce the algorithm in a better way. Also added some references